group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMember.java

```diff
@@ -571,15 +577,29 @@ public ConsumerGroupDescribeResponseData.Member asConsumerGroupDescribeMember(As
     }
 
     private static List<ConsumerGroupDescribeResponseData.TopicPartitions> topicPartitionsFromMap(
-        Map<Uuid, Set<Integer>> partitions
+        Map<Uuid, Set<Integer>> partitions,
+        Map<String, TopicMetadata> subscriptionMetadata
     ) {
         return partitions.entrySet().stream().map(
             item -> new ConsumerGroupDescribeResponseData.TopicPartitions()
                 .setTopicId(item.getKey())
+                .setTopicName(lookupTopicNameById(item.getKey(), subscriptionMetadata))
                 .setPartitions(new ArrayList<>(item.getValue()))
         ).collect(Collectors.toList());
     }
 
+    private static String lookupTopicNameById(
+        Uuid topicId,
+        Map<String, TopicMetadata> subscriptionMetadata
+    ) {
+        for (TopicMetadata topicMetadata : subscriptionMetadata.values()) {
+            if (topicId.equals(topicMetadata.id())) {
+                return topicMetadata.name();
+            }
+        }
```

Comment:
This is pretty bad, isn't it? I was wondering if we could use the MetadataImage instead. It would allow us to get the name based on the id.